New docs (#347)
deepinv/deepinv: docs/source/user_guide/utils.rst

.. _utils:

Utils
=====

Plotting
--------
We provide some plotting functions that are adapted to inverse problems.
The main plotting function is :class:`deepinv.utils.plot`,
which can be used to quickly plot a list of tensor images.

.. doctest::

    >>> from deepinv.utils import plot
    >>> import torch
    >>> x1 = torch.rand(4, 3, 16, 16)
    >>> x2 = torch.rand(4, 3, 16, 16)
    >>> plot([x1, x2], titles=['x1', 'x2'])

We provide other plotting functions that are useful for inverse problems:

.. list-table:: Utility Functions and Descriptions
   :header-rows: 1

   * - **Function**
     - **Description**
   * - :func:`deepinv.utils.plot_curves`
     - Plots curves for visualizing performance metrics over iterations or epochs.
   * - :func:`deepinv.utils.plot_parameters`
     - Visualizes model parameters to analyze distribution and changes over time.
   * - :func:`deepinv.utils.plot_inset`

>      - Plots an image with an inset zoomed view for detailed examination of specific areas.

Reviewer: ```suggestion
     - Plots a list of images with zoomed-in insets extracted from the images.
```

Author: done

   * - :func:`deepinv.utils.plot_videos`

>      - Creates and visualizes videos from sequences of images or frames.

Reviewer: ```suggestion
     - Plots and animates a list of image sequences.
```
